src/ng-select/lib/ng-select.component.ts

```diff
@@ -452,7 +467,7 @@ export class NgSelectComponent implements OnDestroy, OnChanges, OnInit, AfterVie
             return;
         }
 
-        if (this.multiple && item.selected) {
```

Comment:
Without going through deep into this, why is this removed and next line added?

Reply:
I swapped out `this.multiple` for `this.deselectOnClick` because it makes the code more readable.

If `deselectOnClick` wasn't provided to the component, it takes the value of `this.multiple`, and in that case the behaviour remains the same as before. However, if the component received a value for `deselectOnClick`, that value will be used instead of course.